File: clients/javascript/src/models/streamCommitIn.ts
// this file is @generated





export interface StreamCommitIn {
    consumerGroup: string;
offset: number;
topic: string;
}

export const StreamCommitInSerializer = {
    _fromJsonObject(object: any): StreamCommitIn {
        return {
            consumerGroup: object['consumer_group'],
            offset: object['offset'],
            topic: object['topic'],
            };
    },

    _toJsonObject(self: StreamCommitIn): any {
        return {
            'consumer_group': self.consumerGroup,
            'offset': self.offset,
            'topic': self.topic,
            };
    }
}

File: clients/rust/src/models/stream_commit_in.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct StreamCommitIn {
    pub consumer_group: String,

    pub offset: u64,

    pub topic: String,
}

impl StreamCommitIn {
    pub fn new(consumer_group: String, offset: u64, topic: String) -> Self {
        Self {
            consumer_group,
            offset,
            topic,
        }
    }
}
